# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/html/HTMLListAccessible.cpp
# Commit: 854464d3cd64
# Full Hash: 854464d3cd64ec2269638ac7bb0d7bb0f3a1b673
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-02-13 18:21:33
# Regressor Bug: 895323
# File Overlap Count: 1
# Description:
#   Bug 895323 - Don't create bullet accessible when list style is none. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D62537
# ==============================================================================

diff -r ea4df0d263f6 -r 854464d3cd64 accessible/html/HTMLListAccessible.cpp
--- a/accessible/html/HTMLListAccessible.cpp	Thu Feb 13 00:09:16 2020 +0000
+++ b/accessible/html/HTMLListAccessible.cpp	Wed Feb 12 06:25:29 2020 +0000
@@ -38,7 +38,9 @@
     : HyperTextAccessibleWrap(aContent, aDoc), mBullet(nullptr) {
   mType = eHTMLLiType;
 
-  if (nsLayoutUtils::GetMarkerFrame(aContent)) {
+  const nsStyleList* styleList = GetFrame()->StyleList();
+  if (nsLayoutUtils::GetMarkerFrame(aContent) &&
+      (styleList->GetListStyleImage() || !styleList->mCounterStyle.IsNone())) {
     mBullet = new HTMLListBulletAccessible(mContent, mDoc);
     Document()->BindToDocument(mBullet, nullptr);
     AppendChild(mBullet);
